src/python/test/test_datasets_generators.py

```diff
@@ -0,0 +1,39 @@
+""" This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+    Author(s):       Hind Montassif
+
+    Copyright (C) 2021 Inria
+
+    Modification(s):
+      - YYYY/MM Author: Description of the modification
+"""
+
+from gudhi.datasets.generators import points
+
+import pytest
+
+def test_sphere():
+    assert points.sphere(n_samples = 10, ambient_dim = 2, radius = 1., sample = 'random').shape == (10, 2)
+
+    with pytest.raises(ValueError):
+        points.sphere(n_samples = 10, ambient_dim = 2, radius = 1., sample = 'other')
+
+def test_torus():
+    assert points.ctorus(n_samples = 64, dim = 3, sample = 'random').shape == (64, 6)
+    assert points.ctorus(n_samples = 64, dim = 3, sample = 'grid').shape == (64, 6)
+    
+    assert points.ctorus(n_samples = 10, dim = 4, sample = 'random').shape == (10, 8)
+    assert points.ctorus(n_samples = 10, dim = 4, sample = 'grid').shape == (1, 8)
+
+    with pytest.raises(ValueError):
+        points.ctorus(n_samples = 10, dim = 4, sample = 'other')
+
+def test_torus_full_python():
+    assert points.torus(n_samples = 64, dim = 3, sample = 'random').shape == (64, 6)
+    assert points.torus(n_samples = 64, dim = 3, sample = 'grid').shape == (64, 6)
+    
+    assert points.torus(n_samples = 10, dim = 4, sample = 'random').shape == (10, 8)
+    assert points.torus(n_samples = 10, dim = 4, sample = 'grid').shape == (1, 8)
+
+    with pytest.raises(ValueError):
+        points.torus(n_samples = 10, dim = 4, sample = 'other')
```

Comment:
You could factorize your tests this way, and I would add a comment about grid when n_samples < 2**dim:
```suggestion
def _basic_torus(algo):
    assert algo(n_samples = 64, dim = 3, sample = 'random').shape == (64, 6)
    assert algo(n_samples = 64, dim = 3, sample = 'grid').shape == (64, 6)

    assert algo(n_samples = 10, dim = 4, sample = 'random').shape == (10, 8)
    # When n_samples < 2**dim
    assert algo(n_samples = 10, dim = 4, sample = 'grid').shape == (1, 8)

    with pytest.raises(ValueError):
        algo(n_samples = 10, dim = 4, sample = 'other')

def test_torus():
    for torus_algo in [points.torus, points.ctorus]:
        _basic_torus(torus_algo)
```